test/hpke_test.c

```diff
@@ -0,0 +1,1728 @@
+        || base->mode == OSSL_HPKE_MODE_PSKAUTH) {
+        if (!TEST_true(OSSL_HPKE_CTX_set1_psk(sealctx, base->pskid,
+                                              base->psk, base->psklen)))
+            goto end;
+    }
+    if (!TEST_true(OSSL_HPKE_encap(sealctx, enc, &enclen,
+                                   rpub, rpublen,
+                                   base->ksinfo, base->ksinfolen)))
+        goto end;
+    if (!TEST_true(cmpkey(privE, enc, enclen)))
+        goto end;
+    for (i = 0; i < aeadsz; ++i) {
+        ctlen = sizeof(ct);
+        memset(ct, 0, ctlen);
+        if (!TEST_true(OSSL_HPKE_seal(sealctx, ct, &ctlen,
+                                      aead[i].aad, aead[i].aadlen,
+                                      aead[i].pt, aead[i].ptlen)))
+            goto end;
+        if (!TEST_true(TEST_mem_eq(ct, ctlen,
+                                   aead[i].expected_ct,
+                                   aead[i].expected_ctlen)))
+            goto end;
+        if (!TEST_true(OSSL_HPKE_CTX_get_seq(sealctx, &lastseq)))
+            goto end;
+        if (lastseq != (uint64_t)(i + 1))
+            goto end;
+    }
+    if (!TEST_ptr(openctx = OSSL_HPKE_CTX_new(base->mode, base->suite,
+                                              libctx, propq)))
+        goto end;
+    if (base->mode == OSSL_HPKE_MODE_PSK
+        || base->mode == OSSL_HPKE_MODE_PSKAUTH) {
+        if (!TEST_true(base->pskid != NULL && base->psk != NULL
+                       && base->psklen > 0))
+            goto end;
+        if (!TEST_true(OSSL_HPKE_CTX_set1_psk(openctx, base->pskid,
+                                              base->psk, base->psklen)))
+            goto end;
+    }
+    if (base->mode == OSSL_HPKE_MODE_AUTH
+        || base->mode == OSSL_HPKE_MODE_PSKAUTH) {
+        if (!TEST_true(OSSL_HPKE_CTX_set1_authpub(openctx,
+                                                  authpub, authpublen)))
+            goto end;
+    }
+    if (!TEST_true(OSSL_HPKE_decap(openctx, enc, enclen, privR,
+                                   base->ksinfo, base->ksinfolen)))
+        goto end;
+    for (i = 0; i < aeadsz; ++i) {
+        ptoutlen = sizeof(ptout);
+        memset(ptout, 0, ptoutlen);
+        if (!TEST_true(OSSL_HPKE_open(openctx, ptout, &ptoutlen,
+                                      aead[i].aad, aead[i].aadlen,
+                                      aead[i].expected_ct,
+                                      aead[i].expected_ctlen)))
+            goto end;
+        if (!TEST_mem_eq(aead[i].pt, aead[i].ptlen, ptout, ptoutlen))
+            goto end;
+        /* check the sequence is being incremented as expected */
+        if (!TEST_true(OSSL_HPKE_CTX_get_seq(openctx, &lastseq)))
+            goto end;
+        if (lastseq != (uint64_t)(i + 1))
+            goto end;
+    }
+    /* check exporters */
+    if (aeadsz != 0) {
+        /* we already encrypted something with sealctx so better reset seq */
```

Comment:
This looks like a leftover from your original implementation.. Export does not use the seq.

Reply:
True. Deleted that.